File: apps/desktop/src/types/agent.ts
export interface Agent {
    name: string;
    description: string;
    version: string;
    status: "idle" | "standby" | "busy" | "error" | "loading";
    ram_budget_mb: number;
    current_task?: string;
    skills_loaded: string[];
}

export interface AgentStatus {
    name: string;
    status: Agent["status"];
    progress: number;
}

export interface AgentConnection {
    from: string;
    to: string;
    strength: number;
}

File: core/models/agent.py
from pydantic import BaseModel
from typing import Optional


class AgentModel(BaseModel):
    name: str
    description: str
    version: str
    status: str = "idle"
    ram_budget_mb: int
    current_task: Optional[str] = None
    skills_loaded: list[str] = []
